### PR TITLE
Accord: Transaction Expiration

### PR DESCRIPTION
**Pull request by belliottsmith**

### CODE REVIEW

File: src/java/org/apache/cassandra/service/accord/AccordCommandStore.java [hunk 1]
```diff
@@ -321,7 +315,7 @@ public long latestEpoch()
     }
 
     @Override
-    public Timestamp maxConflict(Keys keys)
+    public Timestamp preaccept(TxnId txnId, Keys keys)
```
Comment (aweisberg) [inline src/java/org/apache/cassandra/service/accord/AccordCommandStore.java:318]: Why does `preaccept` not include the same logic as the implementation in `InMemoryCommandStore` around sometimes returning a new `Timestamp`?

File: src/java/org/apache/cassandra/service/accord/api/AccordAgent.java [hunk 2]
```diff
@@ -49,4 +50,10 @@ public void onHandledException(Throwable throwable)
     {
         // TODO: this
     }
+
+    @Override
+    public boolean isExpired(TxnId initiated, long now)
+    {
+        return false;
```
Comment (aweisberg) [inline src/java/org/apache/cassandra/service/accord/api/AccordAgent.java:58]: No check at all? This is TBD until it can be configured?

**Review by aweisberg — CHANGES_REQUESTED** [review]
Had two questions

File: src/java/org/apache/cassandra/service/accord/AccordCommandStore.java (hunk 1 above)
Comment (belliottsmith) [reply src/java/org/apache/cassandra/service/accord/AccordCommandStore.java:318]: Good point. This was a (very) bad rebase.

File: src/java/org/apache/cassandra/service/accord/api/AccordAgent.java (hunk 2 above)
Comment (belliottsmith) [reply src/java/org/apache/cassandra/service/accord/api/AccordAgent.java:58]: Yeah, I think we also want to discriminate between write and read transactions here. I don't mind plumbing it in for now, but it's probably better to leave it until later - it's also not essential for correctness, it only applies to long running original coordinators. I vacillated over whether we should keep it at all, but decided it was probably better.

Comment (belliottsmith) [reply src/java/org/apache/cassandra/service/accord/api/AccordAgent.java:58]: As it happens, it looks like the whole Agent is to-be-implemented anyway

**Review by aweisberg — APPROVED** [review]
+1 TY

**Pull request merged**